packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -70,175 +70,261 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
 }
 
 /// TODO: DOC add docstrings on impl'd methods
-impl<'m> JsPackage<'m> {
+impl<'s> JsPackage<'s> {
     fn import(
-        mv8: &'m MiniV8,
-        embedded: &Embedded<'m>,
+        scope: &mut v8::HandleScope<'s>,
+        context: v8::Local<'s, v8::Context>,
+        embedded: &Embedded<'s>,
         name: &str,
         pkg_type: PackageType,
     ) -> Result<Self> {
         let path = get_pkg_path(name, pkg_type);
-        tracing::debug!("Importing package from path `{}`", &path);
+        tracing::debug!("Importing package from path `{path}`");
         let code = match fs::read_to_string(path.clone()) {
             Ok(s) => s,
             Err(_) => {
                 tracing::debug!("Couldn't read package file. It might intentionally not exist.");
                 // Packages don't have to use JS.
-                let fns = mv8.create_array();
-                fns.set(0, mv8::Value::Undefined)?;
-                fns.set(1, mv8::Value::Undefined)?;
-                fns.set(2, mv8::Value::Undefined)?;
+                let fns = v8::Array::new(scope, 3);
+                let undefined = v8::undefined(scope).into();
+                fns.set_index(scope, 0, undefined)
+                    .ok_or_else(|| Error::V8("Could not create Undefined value".to_string()))?;
```

Comment:
Is that the actual error here? It seems that the error is we're unable to *set* an undefined value within the fns array at index 0

Reply:
Yes, I think the error was on the actual creation at some point but I probably picked the wrong error when refactoring.
Addressed in https://github.com/hashintel/hash/pull/414/commits/2a832f8dc26357b06320aed8e66cb59822d849a2